Cybereason CAR connector

thummalar opened the pull request:
Cybereason CAR connector

thummalar committed:
Cybereason CAR connector

omkar-g commented on connectors/cybereason/configurations/config.json:
```diff
@@ -0,0 +1,26 @@
+    "configuration": {
+        "auth": {
+            "type": "fields",
+            "cybereason_url": {
```
We need to move this to connection, same as: https://github.com/opencybersecurityalliance/stix-shifter/blob/develop/stix_shifter_modules/cybereason/configuration/config.json

We need to follow same structure to that as much as possible.

thummalar replied:
Updated config file as suggested

omkar-g commented on connectors/cybereason/configurations/config.json:
```diff
@@ -0,0 +1,26 @@
+            "password": {
+                "type": "password"
+            },
+            "malop_retention_period": {
```
What is the function of this parameter? Can we just use the schedule value given by user?

thummalar replied:
This is renamed as vulnerability_retention_period, it is used to get the vulnerabilities for the mentioned period ( in days) during incremental update.  It is moved to parameter section.

Commit:
Updated with review comments

1. Updated with review comments for config file changes.
2. Handled incremental delete scenario for asset
3. Updated test json files with dummy values.

thummalar committed:
Merge branch 'IBM:develop' into cybereason

delliott90 commented on connectors/cybereason/configurations/config.json:
```diff
@@ -0,0 +1,37 @@
+                "type": "password"
+            }
+        },
+        "parameter": {
```
To have this more aligned with the UDI connectors config files, I feel this should be under the connection object and instead of named "parameters" change it to "options". You would remove the outer "type" and just leave the "type": "number" under the "vulnerability_retention_period".

thummalar replied:
Updated configuration files as suggested.

Commit:
Addressed review comments, moved parameter fields to connection object's options

Addressed review comments.

Field 'vulnerability_retention_period' moved from parameter object to connection->options object
